### PR TITLE
gate(ROCM): M4 -- pinned vLLM-ROCm oracle on gfx1100 + ROCm device-golden lane in the Qwen3-0.6B near-tie gate

### PR DESCRIPTION
**Pull request by VikashLoomba**
## M4 on gfx1100 — the pinned vLLM-ROCm oracle + the ROCm near-tie gate lane (issue #41)

This PR lands the M4 milestone for `BACKEND-ROCM`: a **pinned vLLM-ROCm oracle built on the same box** (the maintainer's assignment — this is the only board that can host one) and the ROCm arm of the existing device-golden lane in the Qwen3-0.6B near-tie gate.

### The oracle
- Upstream vLLM **at the parity pin `555967922`**, runtime identity `0.23.1rc1.dev1511+g555967922` (verified == `.agents/upstream-sync.md` parity-pin block)
- Built from source as a ROCm wheel for **gfx1100** in `rocm/vllm-dev:base` (2,696 files; `_C`/`_rocm_C`/`_moe_C` ABI extensions present)
- Committed image `vllm-rocm-oracle:555967922-gfx1100`; full recipe in `.agents/specs/rocm-m4-oracle.md`
- **K=10 per-prompt greedy capture: deterministic in every (prompt,pos) cell** — a well-posed strict gate on this board

### The gate result (Qwen3-0.6B, 4x RX 7900 XTX, ROCm 7.14)
```
correctness gate: 16/16 prompts PASS  (STRICT token-exact: 11/16;
near-tie-band only: 5/16; max gap 0.125 nats @ prompt[3] tok=1;
0 forward-divergent) — 125/125 assertions
BACKEND PROOF — Qwen3-dense ops on device type 5 with 0 declines
(kPagedAttention selections=7168)
```
All 28 token-divergent positions vs the oracle's own greedy are inside the 0.5-nat band. The known p0 France/Italy flip is a **literal 0.0000-nat tie in the oracle's own logits** — the Metal M3b signature, reproduced on ROCm: vLLM's teacher-forced argmax on our exact prefix IS `15344 (" Italy")`.

### Change shape (additive only)
- `tests/parity/test_qwen3_paged_engine.cpp`: the existing Metal/Tenstorrent device-golden branch gains the `kROCM` arm — identical anchor + near-tie-band logic, ROCm's own oracle-backed golden pair; **CUDA/CPU/Metal/TT behaviour byte-unchanged**
- New goldens: `our_ids_rocm.npy`, `neartie_gap_mnats_rocm.npy` (the lane pair) + `greedy_ids_rocm.npy`/`greedy_dist_rocm.npy` (oracle capture evidence). **dgx base goldens untouched**
- `.agents/specs/rocm-m4-oracle.md`: the reproducible recipe
- Records: parity-ledger row, owning spec, and STATUS.md. Derived NOW/claim surfaces are intentionally untouched.

### Stacked follow-ups
#570 fixes the 0.8B output-dtype dispatch; #559 adds its green oracle gate; #523 reaches the 35B keep-quant production path. Qwen3-4B strict deterministic gating still needs the checkpoint.

## Current-main integration cutoff (2026-08-18)

Rebased from `17892e7ab7e111a36d70e050bd3b74d984dbc7d3` to `822f4bdece31eedecf71a5fc0fc571e79faa3037` on recorded current-main cutoff `65d6cdaed3e20e9bc70b4f9374fccafefefa7bd0`. Exact M4 0.6B commit replay. Fresh cutoff-head builds, focused gates, full repository preflight, and independent mutation review all pass.

FOLLOWING_AGENTS_PROTOCOL

Following-Agents-Protocol: true
AI-Assisted: true
Assisted-by: pi:kimi-k3 [pi]
Assisted-by: codex:gpt-5.6-sol [codex]
Assisted-by: pi:gpt-5.6-sol [pi]

### CODE REVIEW

**Comment by VikashLoomba** [comment]
## Current-main rebase and fresh review

Rebased from `c3edd8bb` to `ce9ab4ff` on current-main pin `f22c6cc8` and force-updated with an explicit lease.

- Range-diff: exact M4 0.6B patch replay.
- Qwen3-0.6B gate ran on ROCm, not skipped: 16/16 prompts, 125/125 assertions; 11 strict, 5 ratified near-tie, maximum 0.125 nats, zero forward-divergent.
- M4 topology is exact: `f22c6cc8` → this PR → #570 → #559.
- Derived `.agents/NOW.md` and `.agents/coordination.md` are untouched; current-main `head_dim=128` status remains.
- Repository preflight: `All gates green.`
- Fresh independent integration review: **PASS**; final tree clean.

**Comment by VikashLoomba** [comment]
## Final integration cutoff

Final head: `822f4bdece31eedecf71a5fc0fc571e79faa3037`, exact replay on cutoff `65d6cdae`.

- Qwen3-0.6B gate ran on ROCm through `ROCM_ATTN`: 16/16 prompts, 125 assertions; 11 strict, 5 ratified near-tie, max 0.125 nats, zero forward-divergent.
- Final topology is cutoff → this PR → #570 → #559.
- Full preflight and independent cutoff review: **PASS**.
